src/DIRAC/WorkloadManagementSystem/Utilities/Utils.py

```diff
@@ -4,98 +4,53 @@
 import sys
 import json
 
-from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC import gLogger, S_OK
 from DIRAC.Core.Utilities.File import mkDir
-from DIRAC.ConfigurationSystem.Client.Helpers import Registry
-from DIRAC.FrameworkSystem.Client.ProxyManagerClient import gProxyManager
+from DIRAC.FrameworkSystem.private.standardLogging.Logging import Logging
 
 
 def createJobWrapper(
-    jobID,
-    jobParams,
-    resourceParams,
-    optimizerParams,
-    extraOptions="",
-    defaultWrapperLocation="DIRAC/WorkloadManagementSystem/JobWrapper/JobWrapperTemplate.py",
-    log=gLogger,
-    logLevel="INFO",
+    jobID: str,
+    jobParams: dict,
+    resourceParams: dict,
+    optimizerParams: dict,
+    payloadParams: dict | None = None,
+    extraOptions: str | None = None,
+    wrapperPath: str | None = None,
+    rootLocation: str | None = None,
+    pythonPath: str | None = None,
+    defaultWrapperLocation: str | None = "DIRAC/WorkloadManagementSystem/JobWrapper/JobWrapperTemplate.py",
+    log: Logging | None = gLogger,
+    logLevel: str | None = "INFO",
 ):
     """This method creates a job wrapper filled with the CE and Job parameters to execute the job.
-    Main user is the JobAgent
+    Main user is the JobAgent.
+
+    :param jobID: Job ID
+    :param jobParams: Job parameters
+    :param resourceParams: CE parameters
+    :param optimizerParams: Optimizer parameters
+    :param payloadParams: Payload parameters
+    :param extraOptions: Extra options to be passed to the job wrapper
+    :param wrapperPath: Path where the job wrapper will be created
+    :param rootLocation: Location where the job wrapper will be executed
+    :param defaultWrapperLocation: Location of the default job wrapper template
+    :param pythonPath: Path to the python executable
```

Comment:
In the function signature the order is reversed for these two lines
```suggestion
    :param pythonPath: Path to the python executable
    :param defaultWrapperLocation: Location of the default job wrapper template
```